
On OmniOS, several libraries have had their functionality moved into libc
but the old libraries persist as stubs for compatibility reasons. Two such
libraries are libdl.so & libpthread.so

There's no specific problem with an application or library linking with these
(just a slight overhead) but when a library is used by a component within
illumos-omnios, a check is made which generates a warning if an
unnecessary library is linked.

Openssl is one of the libraries used by illumos-omnios and therefore we
patch the configuration script to stop linking against these two libraries.

diff -wpruN '--exclude=*.orig' a~/Configurations/10-main.conf a/Configurations/10-main.conf
--- a~/Configurations/10-main.conf	1970-01-01 00:00:00
+++ a/Configurations/10-main.conf	1970-01-01 00:00:00
@@ -213,7 +213,7 @@ my %targets = (
         inherit_from     => [ "BASE_unix" ],
         template         => 1,
         lib_cppflags     => "-DFILIO_H",
-        ex_libs          => add("-lsocket -lnsl -ldl"),
+        ex_libs          => add("-lsocket -lnsl"),
         dso_scheme       => "dlfcn",
         thread_scheme    => "pthreads",
     },
@@ -241,10 +241,8 @@ my %targets = (
         CC               => "gcc",
         CFLAGS           => add_before(picker(default => "-Wall",
                                               debug   => "-O0 -g",
-                                              release => "-O3 -fomit-frame-pointer")),
-        cflags           => add(threads("-pthread")),
+                                              release => "-O3")),
         lib_cppflags     => add("-DL_ENDIAN"),
-        ex_libs          => add(threads("-pthread")),
         bn_ops           => "BN_LLONG",
         shared_cflag     => "-fPIC",
         shared_ldflag    => add_before("-shared -static-libgcc"),
@@ -265,9 +263,7 @@ my %targets = (
         CFLAGS           => add_before(picker(default => "-Wall",
                                               debug   => "-O0 -g",
                                               release => "-O3")),
-        cflags           => add_before("-m64", threads("-pthread")),
         lib_cppflags     => add("-DL_ENDIAN"),
-        ex_libs          => add(threads("-pthread")),
         bn_ops           => "SIXTY_FOUR_BIT_LONG",
         asm_arch         => 'x86_64',
         perlasm_scheme   => "elf",
